### PR TITLE
Handle h5py dataset timestamps

### PR DESCRIPTION
**Pull request by khl02007**
The timestamps for nwbextractor is of type h5py Dataset which cannot be converted to numpy array by `.asarray()`. I added a line to check the type of the timestamps and use a different method to return it in the case it is a h5py Dataset.

### CODE REVIEW

**Comment by alejoe91** [comment]
Hi @khl02007 

Maybe it's better to directly parse the times in the NwbExtractor?

**Comment by khl02007** [comment]
do you mean override the `get_times` method within the `NwbRecordingExtractor` class? I agree that would be cleaner

**Comment by alejoe91** [comment]
No, I mean forcing to array here: https://github.com/khl02007/spikeinterface/blob/get_times/spikeinterface/extractors/nwbextractors.py#L109

File: spikeinterface/core/baserecording.py [hunk 1]
```diff
@@ -564,8 +564,11 @@ def __init__(self, sampling_frequency=None, t_start=None, time_vector=None):
 
     def get_times(self):
         if self.time_vector is not None:
+            import h5py
             if isinstance(self.time_vector, np.ndarray):
                 return self.time_vector
+            elif isinstance(self.time_vector, h5py.Dataset):
+                return self.time_vector[:]
             else:
                 return self.time_vector.asarray()
```
Comment (samuelgarcia) [inline spikeinterface/core/baserecording.py:573]: I would propose here  `return np.array( self.time_vector)` so no elif and no dep to h5py.
Does it works ?

**Comment by khl02007** [comment]
@samuelgarcia good idea, it seems to work - updated as you suggested.
@alejoe91 what do you think?

**Comment by alejoe91** [comment]
@khl02007 I agree with @samuelgarcia 

This seems to be the best solution!

**Review by alejoe91 — APPROVED** [review]

**Pull request merged**